cpp/src/arrow/array/builder_nested.h

```diff
@@ -80,100 +89,91 @@ class BaseListBuilder : public ArrayBuilder {
     value_builder_->Reset();
   }
 
-  /// \brief Vector append
-  ///
-  /// If passed, valid_bytes is of equal length to values, and any zero byte
-  /// will be considered as a null for that slot
-  Status AppendValues(const offset_type* offsets, int64_t length,
-                      const uint8_t* valid_bytes = NULLPTR) {
-    ARROW_RETURN_NOT_OK(Reserve(length));
-    UnsafeAppendToBitmap(valid_bytes, length);
-    offsets_builder_.UnsafeAppend(offsets, length);
-    return Status::OK();
-  }
-
   /// \brief Start a new variable-length list slot
   ///
   /// This function should be called before beginning to append elements to the
```

Comment:
But `list_util_test.cc` showcases appending on the value builder before calling this...

I think the API contract is ok as written, but then our tests shouldn't abuse the API?

Reply:
This is called to "start a new variable-length list slot". The values added to the values builder before this call are not part of the newly started variable-length list slot.